### ISSUE TITLE
Coverage for all files is not working

### ISSUE THREAD

**Issue by robcaldecott** [vitest-dev/vitest#958]
### Describe the bug

In a React Vite app attempting to show coverage for all files - even those without tests - is not working. This gives misleading statistics about project test coverage.

### Reproduction

The following repo exhibits the issue:

https://github.com/robcaldecott/vitest-coverage-all-files

```
npm install
npm run test:coverage

> vitest-coverage-issue@0.0.0 test:coverage
> vitest run --coverage


 RUN  v0.6.1 /Users/<me>/Projects/vitest-coverage-issue

 √ src/App.test.tsx (1)

Test Files  1 passed (1)
     Tests  1 passed (1)
      Time  5.15s (in thread 30ms, 17004.29%)

----------|---------|----------|---------|---------|-------------------
File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
----------|---------|----------|---------|---------|-------------------
All files |     100 |      100 |      50 |     100 |                   
 App.tsx  |     100 |      100 |      50 |     100 |                   
----------|---------|----------|---------|---------|-------------------
```

The report shows coverage for `App.tsx` but `Button.tsx` and `main.tsx` are not included.

The following Vite config is used:

```ts
/// <reference types="vitest" />

import { defineConfig } from "vite";
import react from "@vitejs/plugin-react";

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [react()],
  test: {
    environment: "jsdom",
    coverage: {
      all: true,
      include: ["src/**/*.tsx"],
    },
  },
});
```

### System Info

```shell
System:
    OS: macOS 10.15.7
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 626.32 MB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 16.13.2 - ~/.nvm/versions/node/v16.13.2/bin/node
    Yarn: 1.22.17 - ~/.nvm/versions/node/v16.13.2/bin/yarn
    npm: 8.5.2 - ~/.nvm/versions/node/v16.13.2/bin/npm
  Browsers:
    Chrome: 99.0.4844.74
    Firefox: 96.0.2
    Safari: 14.1.2
  npmPackages:
    @vitejs/plugin-react: ^1.0.7 => 1.2.0 
    vite: ^2.8.0 => 2.8.6 
    vitest: ^0.6.1 => 0.6.1
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by Demivan** [comment]
Duplicate of #680

**Comment by larsthorup** [comment]
Note that while #680 is still open, the issue reported here with .tsx files missing from the coverage report is now fixed with the latest version of c8@7.11.3 - I was able to get full coverage after running `npm upgrade c8`.
